### PR TITLE
SISRP-13242 Incorporate Hub academic status data into My Academics profile

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.berkeley.edu/browse/SISRP-13242

As directed, we continue to use legacy sources for eight-digit IDs and switch to the Hub for ten-digit IDs. Loath as I am to create one more feature flag, it seems prudent to give us an emergency switchoff if needed.

Since our test data currently lacks variety, minors are not yet handled, and cases such as double majors are not yet tested.

The length-based determination of whether an ID counts as legacy is rehomed in the User::Student module. The module is also nudged toward more frequent caching of looked-up CS IDs, without a full-on refactor.

### CODE REVIEW

File: app/models/user/student.rb [hunk 1]
```diff
@@ -16,6 +20,11 @@ def lookup_delegate_user_id
       CalnetCrosswalk::ByUid.new(user_id: @uid).lookup_delegate_user_id
     end
 
+    def legacy_campus_solutions_id?
```
Comment (johncrossman) [inline app/models/user/student.rb]: @paulkerschen without the `?` I'd assume this a "getter" method. Would `has_legacy_campus_solutions_id?` be closer to conventional naming? (And more in line w/ `@edo_attributes` key?)

File: spec/models/my_academics/gpa_units_spec.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
+describe 'MyAcademics::GpaUnits' do
+
```
Comment (johncrossman) [inline spec/models/my_academics/gpa_units_spec.rb]: Nice overall test coverage, @paulkerschen

**Comment by johncrossman** [comment]
Looks good, @paulkerschen  Good to merge once you comment on the above.

File: app/models/user/student.rb (hunk 1 above)
Comment (raydavis) [inline app/models/user/student.rb]: I think the Ruby convention is that the "?" eliminates the need for a "has_" or "is_", although I'm often tempted to go against it.

Comment (raydavis) [inline app/models/user/student.rb]: Maybe `legacy_student?` would convey the functional intention?

**Comment by raydavis** [comment]
Aside from the evergreen topic of naming, this looks like a coherent-as-possible depiction of a well-muddied situation. Thanks for wading through it, @paulkerschen .

File: app/models/user/student.rb (hunk 1 above)
Comment (paulkerschen) [inline app/models/user/student.rb]: The only problem with `legacy_student?` as a method name is the same problem with `User::Student` as a module name (and "student EDO" as our data source...); that it applies to everyone, not just students. I foresee a future refactor trying to mitigate this.

Anyhow, I think @johncrossman's more explicit `has_legacy_campus_solutions_id?` is reasonable, and will amend.

Comment (paulkerschen) [inline app/models/user/student.rb]: Or, more concisely, perhaps `legacy_user?` with a corresponding `:isLegacyUser` key cuts to the chase.

**Comment by paulkerschen** [comment]
Thanks for conscientious review, @johncrossman and @raydavis. Travis has blessed an amendment with `legacy_user?` method name and `:is_legacy_user` attributes key.

**Comment by raydavis** [comment]
Wouldn't `valued_customer?` be friendlier?

**Pull request merged**